Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -26,6 +26,9 @@ public enum VersionSetSpecifier: Equatable {
     /// A non-empty range of version.
     case range(Range<Version>)
 
+    /// The unit or singleton set.
+    case unit(Version)
```

Comment:
Thanks for working on this!

I wonder if it makes sense to specify a unit version, or if makes more sense to specify a "fixed" dependency where the resolver doesn't actually know about the version, it just knows about what the dependencies at that point are?

In the case when we have a version pinned at a SHA, then this won't cover that case, and it feels like we should be able to cover both cases with the same enhancement.
In other words,